src/mnode/rpc/masternode.cpp

```diff
@@ -55,11 +55,20 @@ UniValue masternodelist(const UniValue& params, bool fHelp)
 {
     std::string strMode = "status";
     std::string strFilter;
+    std::string strExtra = "";
 
     if (!params.empty())
         strMode = params[0].get_str();
-    if (params.size() == 2)
+    if (params.size() >= 2)
         strFilter = params[1].get_str();
+    if (params.size() == 3)
+    {
+        strExtra = params[2].get_str();
+    }
+    else if (params.size() > 3)
+    {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Too many parameters");
+    }
 
     if (fHelp || (strMode != "activeseconds" && strMode != "addr" && strMode != "full" && strMode != "info" &&
```

Comment:
can you change this to RPC_CMD_PARSER? (look at any mnode/rpc/ticket-*.cpp files for example)